lib/parseInputOrigins.js

```diff
@@ -17,16 +23,31 @@ module.exports = (origins) => {
     if (typeof origin === 'object') {
       // add any cache behaviors
       for (const pathPattern in origin.pathPatterns) {
-        const cacheBehavior = getCacheBehavior(
-          pathPattern,
-          origin.pathPatterns[pathPattern],
-          originConfig.Id
-        )
+        const pathPatternConfig = origin.pathPatterns[pathPattern]
+        const cacheBehavior = getCacheBehavior(pathPattern, pathPatternConfig, originConfig.Id)
+
+        const lambdaAtEdge = pathPatternConfig['lambda@edge'] || {}
+
+        Object.keys(lambdaAtEdge).forEach((eventType) => {
+          if (!validLambdaTriggers.includes(eventType)) {
+            throw new Error(
+              `"${eventType}" is not a valid lambda trigger. See https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-cloudfront-trigger-events.html for valid event types.`
+            )
+          }
+
+          cacheBehavior.LambdaFunctionAssociations.Quantity = 1
```

Comment:
do you think it would be limiting to just have a single lambda association? 🤔 ... could you think of some use cases that would require more?

Reply:
Nice catch! I've updated the tests and impl to set the correct Quantity. Yes, users should be allowed to set up to 4 associations for the 4 different event types.